var/spack/repos/builtin/packages/bufr/package.py

```diff
@@ -21,3 +21,37 @@ class Bufr(CMakePackage):
                    'jbathegit']
 
     version('11.5.0', sha256='d154839e29ef1fe82e58cf20232e9f8a4f0610f0e8b6a394b7ca052e58f97f43')
+
+    def setup_run_environment(self, env):
+
+        lib4 = find_libraries('libbufr_4', root=self.prefix,
+                              shared=False, recursive=True)
+        lib8 = find_libraries('libbufr_8', root=self.prefix,
+                              shared=False, recursive=True)
+        libd = find_libraries('libbufr_8', root=self.prefix,
+                              shared=False, recursive=True)
```

Comment:
Is this a typo:
```suggestion
        libd = find_libraries('libbufr_d', root=self.prefix,
                              shared=False, recursive=True)
```
?

Reply:
Can we factor together the operations for a single library? For instance:
```python
def _setup_libufr_environment(self, env, suffix):
    libname = 'libufr_{0}'.format(suffix)
    ...

def setup_run_environment(self, env):
    for suffix in ('4', '8', 'd'):
        self._setup_libufr_environment(env, suffix)
```

Reply:
Yes

Reply:
Done, how does it look now?